### PR TITLE
CLOUD-817 Correctly determine network address in Openstack

### PR DESCRIPTION
**Pull request by akanto**
@biharitomi or @keyki please check

- OpenStack requires floatingips only for gateway

### CODE REVIEW

File: core/src/main/java/com/sequenceiq/cloudbreak/service/stack/connector/openstack/HeatTemplateBuilder.java [hunk 1]
```diff
@@ -11,25 +11,34 @@
...
 import freemarker.template.TemplateException;
+import jersey.repackaged.com.google.common.collect.Maps;
 
 @Service
 public class HeatTemplateBuilder {
 
     public static final String CB_INSTANCE_GROUP_NAME = "cb_instance_group_name";
     public static final String CB_INSTANCE_PRIVATE_ID = "cb_instance_private_id";
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(OpenStackConnector.class);
```
Comment (doktoric) [inline core/src/main/java/com/sequenceiq/cloudbreak/service/stack/connector/openstack/HeatTemplateBuilder.java]: HeatTemplateBuilder.class

File: core/src/main/java/com/sequenceiq/cloudbreak/service/stack/connector/openstack/HeatTemplateBuilder.java [hunk 2]
```diff
@@ -70,18 +82,30 @@ public String add(Stack stack, String gateWayUserData, String coreUserData, Set<
...
+            model.put("cb_stack_name", stack.getName());
             model.put("agents", agents);
             model.put("coreuserdata", formatUserData(coreUserData));
             model.put("gatewayuserdata", formatUserData(gateWayUserData));
             model.put("subnets", stack.getAllowedSubnets());
             model.put("ports", NetworkUtils.getPorts(stack));
-            model.put("cbSubnet", stack.getNetwork().getSubnetCIDR());
-            return processTemplateIntoString(freemarkerConfiguration.getTemplate(openStackHeatTemplatePath, "UTF-8"), model);
+            String generatedTemplate = processTemplateIntoString(freemarkerConfiguration.getTemplate(openStackHeatTemplatePath, "UTF-8"), model);
+            LOGGER.debug("Generated Heat template: {}" + generatedTemplate);
```
Comment (doktoric) [inline core/src/main/java/com/sequenceiq/cloudbreak/service/stack/connector/openstack/HeatTemplateBuilder.java]: no need for + sign

Comment (akanto) [inline core/src/main/java/com/sequenceiq/cloudbreak/service/stack/connector/openstack/HeatTemplateBuilder.java]: fixed

File: core/src/main/java/com/sequenceiq/cloudbreak/service/stack/connector/openstack/HeatTemplateBuilder.java (hunk 1 above)
Comment (akanto) [inline core/src/main/java/com/sequenceiq/cloudbreak/service/stack/connector/openstack/HeatTemplateBuilder.java]: thanks, fixed

**Pull request merged**